donkeycar/management/makemovie.py

```diff
@@ -213,10 +214,9 @@ def draw_salient(self, img):
 
         norm_img = normalize_image(img)
         salient_mask = self.compute_visualisation_mask(norm_img)
-        z = np.zeros_like(salient_mask)
-        salient_mask_stacked = np.dstack((z, z))
-        salient_mask_stacked = np.dstack((salient_mask_stacked, salient_mask))
-        blend = cv2.addWeighted(img.astype('float32'), alpha, salient_mask_stacked, beta, 0.0)
+        salient_mask_stacked = cm.inferno(salient_mask)[:,:,0:3]
+        salient_mask_stacked = cv2.GaussianBlur(salient_mask_stacked,(3,3),cv2.BORDER_DEFAULT)
```

Comment:
Is using the Gaussian Blur here the reason we are not seeing the respective pixels of the mask as color variations?

Reply:
The `cv2.GaussianBlur` is to smooth the saliency map and hence gives a betterlooking visualization. 

I would like to clarify two points.
- First of all, please note that the original codes represent the saliency map with blue color as the following lines shown https://github.com/autorope/donkeycar/blob/20bf83c4f8bfe24bda151659e271a488bfa91f1b/donkeycar/management/makemovie.py#L216-L218 but I represent it using a colormap `inferno` (black-purple-yellow) https://github.com/robocarstore/donkeycar/blob/d5226637b668df534c680628e8c040b66e60d22a/donkeycar/management/makemovie.py#L217 . ![saliency compare](https://user-images.githubusercontent.com/80034477/122859144-19e0ba00-d34e-11eb-91f3-ace08d8001f6.png)
(left): saliency represented by inferno colormap (right): saliency represented by blue color

- Secondly, the sharp red, yellow, blue colors of the video produced by the original codes are not saliency map but artifacts due to the overflow of numpy uint8 conversion. 
![Drawing (1)](https://user-images.githubusercontent.com/80034477/122859769-1ac61b80-d34f-11eb-9f97-1d1ed522cb96.png)

Reply:
Ok, that was a severe bug then - so the original salient wasn't highlighting the activations but just showing overflowing bright pixels. Great that you discovered that. That's a big improvement.